src/main/kotlin/com/sourcegraph/cody/history/listener/RightClickListener.kt

```diff
@@ -0,0 +1,12 @@
+package com.sourcegraph.cody.history.listener
+
+import java.awt.event.MouseAdapter
+import java.awt.event.MouseEvent
+import javax.swing.SwingUtilities
+
+class RightClickListener(private val action: (MouseEvent) -> Unit) : MouseAdapter() {
+
+  override fun mousePressed(e: MouseEvent?) {
+    if (SwingUtilities.isRightMouseButton(e)) action(e!!)
```

Comment:
Again, no guarantees it will work I guess? Maybe we can reflect the reality by changing action to be `private val action: (MouseEvent?) -> Unit`? I must check the use side.